python/cudf_polars/cudf_polars/experimental/select.py

```diff
@@ -27,10 +58,90 @@ def _(
     if pi.count > 1 and not all(
         expr.is_pointwise for expr in traversal([e.value for e in ir.exprs])
     ):
-        # TODO: Handle non-pointwise expressions.
-        raise NotImplementedError(
-            f"Selection {ir} does not support multiple partitions."
-        )
+        try:
+            # Try decomposing the underlying expressions
+            return decompose_select(ir, child, partition_info)
+        except NotImplementedError:
+            # TODO: Handle more non-pointwise expressions.
+            return _lower_ir_fallback(
+                ir, rec, msg="This selection is not supported for multiple partitions."
+            )
+
     new_node = ir.reconstruct([child])
     partition_info[new_node] = pi
     return new_node, partition_info
+
+
+def build_fusedexpr_select_graph(
+    ir: Select, partition_info: MutableMapping[IR, PartitionInfo]
+) -> MutableMapping[Any, Any]:
+    """Build complex Select graph."""
+    (child,) = ir.children
+    child_count = partition_info[child].count
+
+    # Build Graph to produce a column for each
+    # NamedExpr element of ir.exprs
+    graph: MutableMapping[Any, Any] = {}
+    expr_partition_counts: MutableMapping[Expr, int] = {}
+    roots = []
+    for ne in ir.exprs:
+        assert isinstance(ne.value, FusedExpr), f"{ne.value} is not a FusedExpr"
+        roots.append(ne)
+        expr_partition_counts = extract_partition_counts(
+            [ne],
+            child_count,
+            update=expr_partition_counts,
+        )
+        for node in traversal([ne.value]):
+            assert isinstance(node, FusedExpr), f"{node} is not a FusedExpr"
```

Comment:
This is checking things that are asserted in the constructor at least because here we've already checked we have a `FusedExpr`.